packages/patterns/src/types.ts

```diff
@@ -46,3 +46,50 @@ export const PatternPropType = PropTypes.shape({
     h: PropTypes.number.isRequired,
   }),
 });
+
+export enum PatternType {
+  SOLID = 'solid',
+  LINEAR = 'linear',
+  RADIAL = 'radial',
+}
+
+export type Hex = {
+  r: number;
+  g: number;
+  b: number;
+  a?: number;
+};
+
+export type Solid = {
+  type?: PatternType.SOLID;
+  color: Hex;
+};
+
+export type ColorStop = {
+  color: Hex;
+  position: number;
+};
+
+export type Linear = {
+  type: PatternType.LINEAR;
+  stops: Array<ColorStop>;
+  rotation?: number;
+  alpha?: number;
+};
+
+export type Radial = {
+  type: PatternType.RADIAL;
+  stops: Array<ColorStop>;
+  size?: {
+    w: number;
+    h: number;
+  };
+  center?: {
+    x: number;
+    y: number;
+  };
+  rotation?: number;
+  alpha?: number;
+};
+
+export type Pattern = Solid | Linear | Radial;
```

Comment:
Curious why use types and not interfaces for the objects? As far as I've read so far the general recommendation seems to be using types for functions and interfaces for objects, however, it looks like with the latest TS versions, these two have become more similar. Is there any reason why you prefer types to interfaces here?

Reply:
They're mostly interchangeable and I don't think there's any real difference in these cases. Interfaces are easier to extend, but this doesn't require extension ever, so not require. The union type though can only be a type (you can't create a union interface IIRC), so the final thing here has to be type, but the other things could be interfaces.

Reply:
I did improve the code around this a bit though, creating a `Gradient` intermediate type. Not sure if this is best-practice, but it works out pretty well.